contracts/Safe.sol

```diff
@@ -444,8 +400,41 @@ contract Safe is
         address gasToken,
         address refundReceiver,
         uint256 _nonce
-    ) public view override returns (bytes32) {
-        return keccak256(encodeTransactionData(to, value, data, operation, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver, _nonce));
+    ) public view override returns (bytes32 txHash) {
+        bytes32 separatorHash = domainSeparator();
+        /* solhint-disable no-inline-assembly */
```

Comment:
nit: Can you add a comment here justifying the use of inline assembly? I think it makes sense to add these kinds of justifications so that auditors and other people reviewing the code can understand our rationale behind not just using plain Solidity for this.